libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -211,6 +240,24 @@ static int start_c11(void *p)
 	__pthread_exit((void *)(uintptr_t)start(args->start_arg));
 	return 0;
 }
+#else
+__attribute__((export_name("wasi_thread_start")))
+int wasi_thread_start(int tid, void *p)
```

Comment:
Because if we pass out a function pointer, then in order to have it be useable/callable, we also need to export the entire table from the wasm module.   We discussed the the cost/benefit of doing this (sorry I can't remember exactly where).    

My personal feeling is that exporting the entire table just for this purpose is not worth it.   In addition of over-sharing, there is also at least one binaryen optimization that is prevented when the table is exported (The important detail is that a non-exported table can be considered read-only when there are no table.set instructions and binaryen can do it calls "directize"ing in this case.. i.e. it can turn some indirect calls into direct calls).

Reply:
I guess once we have first class function references then maybe we could switch to using them instead of a fixed export?  (i.e. we could do table.get and then pass the result out of the host).

Reply:
If we pass a function pointer, then the Wasm module must export its function `table` under a fixed name so the host can access the table the function pointer lives in. What's the difference between exporting a function and a table? I can't recall all the details from our last discussion, but IIRC it was trickier to export the function table (toolchain-wise) and, in my opinion, just setting a function name is a bit more direct and clear.

Reply:
(@sbc100, seems like we raced!)